specification/devcenter/DevCenter/DevBox/models.tsp

```diff
@@ -0,0 +1,446 @@
+  @visibility("read")
+  osBuildNumber?: string;
+
+  @doc("The datetime that the backing image version was published.")
+  @visibility("read")
+  publishedDate?: utcDateTime;
+}
+
+@doc("Stop on disconnect configuration settings for Dev Boxes created in this pool.")
+model StopOnDisconnectConfiguration {
+  @doc("""
+Indicates whether the feature to stop the devbox on disconnect once the grace
+period has lapsed is enabled.
+""")
+  status: StopOnDisconnectEnableStatus;
+
+  @doc("""
+The specified time in minutes to wait before stopping a Dev Box once disconnect
+is detected.
+""")
+  gracePeriodMinutes?: int32;
+}
+
+@doc("A Schedule to execute action.")
+@resource("schedules")
+@parentResource(Pool)
+model Schedule {
+  @key("scheduleName")
+  @visibility("read")
+  @doc("Display name for the Schedule")
+  name: string;
+
+  @doc("Supported type this scheduled task represents.")
+  type: ScheduledType;
+
+  @doc("The frequency of this scheduled task.")
+  frequency: ScheduledFrequency;
+
+  @doc("The target time to trigger the action. The format is HH:MM.")
+  //no build in time type matches HH:MM time format
+  time: string;
+
+  @doc("The IANA timezone id at which the schedule should execute.")
+  timeZone: string;
+}
+
+@doc("The Dev Box list result")
+model DevBoxListResult is Azure.Core.Page<DevBox>;
+
+@doc("A Dev Box")
+@resource("devboxes")
+@parentResource(User)
+model DevBox {
+  @key("devBoxName")
+  @doc("Display name for the Dev Box")
+  @minLength(3)
+  @maxLength(63)
+  @pattern("^[a-zA-Z0-9][a-zA-Z0-9-_.]{2,62}$")
+  @visibility("read")
+  name: string;
+
+  @doc("Name of the project this Dev Box belongs to")
+  @visibility("read")
+  projectName?: string;
+
+  @doc("The name of the Dev Box pool this machine belongs to.")
+  @minLength(3)
+  @maxLength(63)
+  @pattern("^[a-zA-Z0-9][a-zA-Z0-9-_.]{2,62}$")
+  @visibility("read", "create")
+  poolName: string;
+
+  @doc("Indicates whether hibernate is enabled/disabled or unknown.")
+  @visibility("read")
+  hibernateSupport?: HibernateSupport;
+
+  @doc("The current provisioning state of the Dev Box.")
+  @visibility("read")
+  provisioningState?: DevBoxProvisioningState;
+
+  @doc("""
+The current action state of the Dev Box. This is state is based on previous
+action performed by user.
+""")
+  @visibility("read")
+  actionState?: string;
+
+  @doc("The current power state of the Dev Box.")
+  @visibility("read")
+  powerState?: PowerState = PowerState.Unknown;
+
+  @doc("""
+A unique identifier for the Dev Box. This is a GUID-formatted string (e.g.
+00000000-0000-0000-0000-000000000000).
+""")
+  @visibility("read")
+  uniqueId?: Azure.Core.uuid;
+
+  @doc("Provisioning or action error details. Populated only for error states.")
+  @visibility("read")
+  error?: Azure.Core.Foundations.Error;
+
+  @doc("""
+Azure region where this Dev Box is located. This will be the same region as the
+Virtual Network it is attached to.
+""")
+  @visibility("read")
+  location?: string;
```

Comment:
Let's update to the `azureLocation` type in tsp azure core: https://azure.github.io/typespec-azure/docs/libraries/azure-core/reference/data-types#Azure.Core.azureLocation
```suggestion
  location?: azureLocation;
```